xdsl/irdl/attributes.py

```diff
@@ -233,8 +212,14 @@ def irdl_param_attr_definition(cls: _PAttrTT) -> _PAttrTT:
     new_fields = get_accessors_from_param_attr_def(attr_def)
 
     if issubclass(cls, TypedAttribute):
-        parameter_names: tuple[str] = tuple(zip(*attr_def.parameters))[0]
-        type_index = parameter_names.index("type")
+        type_indexes = tuple(
+            i for i, (p, _) in enumerate(attr_def.parameters) if p == "type"
+        )
+        if not type_indexes:
+            raise PyRDLAttrDefinitionError(
+                f"TypedAttribute {cls.__name__} should have a 'type' parameter."
+            )
+        type_index = type_indexes[0]
```

Comment:
Should this logic be put as a function in `TypedAttribute` instead?
By having an accessor to `type` directly with something like `get_type` for instance?

Reply:
Are you suggesting to remove `get_type_index` entirely?

Reply:
The issue is that I think you want this check to happen at the definition of the attribute, rather than some verify check on an instance of that attribute

Reply:
actually nevermind I'm an idiot, this is all good!